src/test/java/org/broadinstitute/hellbender/utils/pca/PCAUnitTest.java

```diff
@@ -139,6 +141,50 @@ public void testSparkSVDOnTestMatrix() {
         assertEqualEigenVectors(eigenVectors, new Array2DRowRealMatrix(TEST_EXPECTED_EIGENVECTORS), variances, EPSILON);
     }
 
+    @Test(dependsOnMethods = "testResidentSVDOnTestMatrix")
+    public void testReadHDF5() throws IOException {
+        final RealMatrix dataMatrix = new Array2DRowRealMatrix(TEST_MATRIX);
+        final List<String> variableNames = createVariableNames(dataMatrix.getRowDimension());
+        final List<String> sampleNames = createSampleNames(dataMatrix.getColumnDimension());
+        final PCA pca = PCA.createPCA(variableNames, sampleNames, dataMatrix, SVDFactory::createSVD);
+        final File outputFile = createTempFile("pca-test", ".hd5");
+        final File outputFile2 = createTempFile("pca-test-2", ".hd5");
+        try (final HDF5File outFile = new HDF5File(outputFile, HDF5File.OpenMode.CREATE)) {
+            PCA.writeHDF5(pca, outFile);
+        }
+        // We will read out of a copy to make sure there is caching magic playing part.
```

Comment:
not sure if i understand this comment...did you mean to say "no caching magic taking part"?  same below.